src/NuGetForUnity/Editor/NugetHelper.cs

```diff
@@ -1251,12 +1295,11 @@ public static bool Install(NugetPackage package, bool refreshAssets = true)
                         foreach (var entry in zip.Entries)
                         {
                             var filePath = Path.Combine(baseDirectory, entry.FullName);
-                            var directory = Path.GetDirectoryName(filePath);
-                            Directory.CreateDirectory(directory);
-                            if (Directory.Exists(filePath))
-                            {
-                                continue;
-                            }
+
+                            var isDir = filePath.EndsWith("/") || filePath.EndsWith("\\");
```

Comment:
It seems you are right. We got confused since the old code said `if (Directory.Exists(filePath))` which then doesn't make sense either since filePath, as you said can't be a directory. Should that just be fixed to `if (File.Exists(filePath))` then, or just removed?

Reply:
I think `if (Directory.Exists(filePath))` is a workaround if the Zip contains a file with the exact same path as a directory that already exists on the HDD. `entry.ExtractToFile(filePath, true);` only overwrites files that already exists but if it is a dictionary it would lead to a exception. I think the best would be to keep the 
```
if (Directory.Exists(filePath))
{
    continue;
}
```
but add a Log warning, because overwriting a directory can lead to "data loos".

Reply:
We just found a package where some entries are actually directories, where entry.FullName ends with '/': Microsoft.Azure.WebJobs.Sources

It is causing `The path ...\contentFiles\ refers to an existing directory. Overwriting it may lead to data loss.` warning. We will try to fix this within the next pull request we make. @JoC0de I just wanted to let you know this exists.

Reply:
Really interesting, what exactly the entry is than? Because at least it could have a stream / size, can we check this? Because it doesn't seem relatable to check the "fileName".

Reply:
entry.FullName ends with '/' but entry.Name is empty and Length and CompressedLength are zero. Those are all useful properties on ZipEntry. Length and CompressedLength will be 0 for regular files that are empty so I guess the best check we can do is `string.IsNullOrEmpty(entry.Name)`.